session.go

```diff
@@ -768,12 +768,30 @@ func (s *session) processTransportParameters(params *handshake.TransportParamete
 }
 
 func (s *session) sendPackets() error {
+	s.packer.SetLeastUnacked(s.sentPacketHandler.GetLeastUnacked())
 	s.pacingDeadline = time.Time{}
 	if !s.sentPacketHandler.SendingAllowed() { // if congestion limited, at least try sending an ACK frame
 		return s.maybeSendAckOnlyPacket()
 	}
 	numPackets := s.sentPacketHandler.ShouldSendNumPackets()
-	for i := 0; i < numPackets; i++ {
+	var numPacketsSent int
+	// Send retransmissions, until
+	// * we're congestion limited, or
+	// * there are no more retransmissions, or
+	// * the maximum number of packets was reached
+	for ; numPacketsSent < numPackets; numPacketsSent++ {
+		sentPacket, err := s.maybeSendRetransmission()
+		if err != nil {
+			return err
+		}
+		if !sentPacket { // no more retransmission to send. Proceed to send new data.
+			break
+		}
+		if !s.sentPacketHandler.SendingAllowed() {
```

Comment:
Is this correct? I vaguely remember that we wanted to allow retransmissions even when congestion limited?

Reply:
That's handled by https://github.com/lucas-clemente/quic-go/blob/master/internal/ackhandler/sent_packet_handler.go#L384-L388.
I'm currently refactoring the logic of `SendingAllowed`, to make it less hacky, and I'll send a PR later this week.